src/renderer/modals/FullNode/FullNodeBody.js

```diff
@@ -0,0 +1,167 @@
+import Breadcrumb from "~/renderer/components/Stepper/Breadcrumb";
+import StepLanding, { StepLandingFooter } from "./steps/StepLanding";
+import StepNode, { StepNodeFooter } from "./steps/StepNode";
+import StepAccounts, { StepAccountsFooter } from "./steps/StepAccounts";
+import StepConnectDevice, { StepDeviceFooter } from "./steps/StepConnectDevice";
+import StepSatStack, { StepSatStackFooter } from "./steps/StepSatStack";
+import StepDisconnect, { StepDisconnectFooter } from "./steps/StepDisconnect";
+import type { FullNodeSteps, ConnectionStatus } from "~/renderer/modals/FullNode";
+import { connectionStatus } from "~/renderer/modals/FullNode";
+import { saveLSS } from "./helpers";
+import { getEnv } from "@ledgerhq/live-common/lib/env";
+import { loadLSS } from "~/renderer/modals/FullNode/helpers";
+
+const steps = ["landing", "node", "accounts", "device", "satstack", "disconnect"];
+const FullNodeBody = ({
+  onClose,
+  onStepChange,
+  activeStep,
+}: {
+  onClose: any,
+  onStepChange: FullNodeSteps => void,
+  activeStep: FullNodeSteps,
+}) => {
+  const [errors, setErrors] = useState(null);
+  const satStackAlreadyConfigured = getEnv("SATSTACK");
+  const [satStackDownloaded, setSatStackDownloaded] = useState(false);
+
+  const [nodeConnectionStatus, setNodeConnectionStatus] = useState<ConnectionStatus>(
+    connectionStatus.IDLE,
+  );
+  const items = [
+    { label: <Trans i18nKey={"fullNode.modal.steps.node.title"} /> },
+    { label: <Trans i18nKey={"fullNode.modal.steps.accounts.title"} /> },
+    { label: <Trans i18nKey={"fullNode.modal.steps.device.title"} /> },
+    { label: <Trans i18nKey={"fullNode.modal.steps.satstack.title"} /> },
+  ];
+
+  const [numberOfAccountsToScan, setNumberOfAccountsToScan] = useState(10);
+  const [nodeConfig, setNodeConfig] = useState<RPCNodeConfig>({
+    host: "",
+    username: "",
+    password: "",
+    notls: false,
+  });
+  const [scannedDescriptors, setScannedDescriptors] = useState();
+  const patchNodeConfig = useCallback((patch: any) => setNodeConfig({ ...nodeConfig, ...patch }), [
+    nodeConfig,
+  ]);
+
+  const errorSteps = errors ? [1] : [];
+  const showBreadcrumb = !["landing", "disconnect"].includes(activeStep);
+  const stepIndex = steps.indexOf(activeStep) - 1;
+
+  const onAcceptLanding = useCallback(() => onStepChange("node"), [onStepChange]);
+
+  const validNodeConfig = useMemo(() => {
+    const errors = validateRPCNodeConfig(nodeConfig);
+    // NB All fields are required, why do we need these errors?
+    return !errors.length;
+  }, [nodeConfig]);
+
+  useEffect(() => {
+    // Prefill the node data for edit?
+    async function prefill() {
+      if (satStackAlreadyConfigured) {
+        const maybeNodeConfig = await loadLSS();
+        if (maybeNodeConfig && maybeNodeConfig.node) {
+          setNodeConfig(maybeNodeConfig.node);
+        }
+      }
+    }
+    prefill();
+  }, [satStackAlreadyConfigured]);
+
+  useEffect(() => {
+    if (scannedDescriptors) {
```

Comment:
the condition feels a bit weird.
this sounds like it could run if you edit nodeConfig and have scannedDescriptors settled. But i guess it works since scannedDescriptors changes as second step.
But then, what about the Edit mode? Should we save it when user just "fixes" its config but without needing to scan any new descriptors?

In term of flow, I have feeling that we have two places where we should save:
- after full config (node + descriptors) (for both first time and re-edit time)
- after node config connection succeed validation and **if it's edit mode** (🤔  even tho we could allow to save it even on the first run, but it would make the set up partially complete (missing accounts descriptor) so probably to just do it in edit mode)

Reply:
it will save in both steps after initial setup and only after descriptor import on the initial setup